server/storage.go

```diff
@@ -561,3 +564,113 @@ func saveGDriveToken(path string, token *oauth2.Token, logger *log.Logger) {
 
 	json.NewEncoder(f).Encode(token)
 }
+
+var uplinkFailure = errs.Class("storj failure")
+
+type StorjStorage struct {
+	Storage
+	uplink  *uplink.Uplink
+	project *uplink.Project
+	bucket  *uplink.Bucket
+	logger  *log.Logger
+}
+
+func NewStorjStorage(endpoint, apiKey, bucket, encKey string, logger *log.Logger) (*StorjStorage, error) {
+	var instance StorjStorage
+	var err error
+
+	ctx := context.TODO()
+
+	instance.uplink, err = uplink.NewUplink(ctx, nil)
+	if err != nil {
+		return nil, uplinkFailure.Wrap(err)
```

Comment:
no real needs in my opinion for all these errors wrapping. I would rather return the given `err` as returned by the sdk everywhere